tiledb/ml/models/_base.py

```diff
@@ -212,32 +197,63 @@ def _serialize_tensorboard_files(log_dir: str = "") -> bytes:
+                )
+
+            md_contents = model_array[0:model_size]["model"]
+            return pickle.loads(md_contents.tobytes())  # type: ignore
 
+    def get_optimizer_weights(self, timestamp: Optional[Timestamp] = None) -> Weights:
+        """
+        Returns optimizer's weights. Works for Tensorflow Keras and PyTorch
+        """
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
             try:
-                tensorboard_size = meta["tensorboard_size"]
+                optimizer_size = model_array.meta["optimizer_size"]
+            except KeyError:
+                raise Exception(
+                    f"optimizer_size metadata entry not present in {self.uri}"
+                    f" (existing keys: {set(model_array.meta.keys())})"
+                )
+
+            opt_contents = model_array[0:optimizer_size]["optimizer"]
+            return pickle.loads(opt_contents.tobytes())  # type: ignore
+
+    def _load_tensorboard(self, timestamp: Optional[Timestamp] = None) -> None:
+        """
+        Writes Tensorboard files to directory. Works for Tensorflow-Keras and PyTorch.
+        """
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            try:
+                tensorboard_size = model_array.meta["tensorboard_size"]
             except KeyError:
                 raise Exception(
                     f"tensorboard_size metadata entry not present in"
-                    f" (existing keys: {set(meta)})"
+                    f" (existing keys: {set(model_array.meta.keys())})"
                 )
 
-            if tensorboard_size:
-                tensorboard_contents: np.ndarray = model_array[0:tensorboard_size][
-                    "tensorboard"
-                ]
-                tensorboard_files = pickle.loads(tensorboard_contents.tobytes())
-
-                for path, file_bytes in tensorboard_files.items():
-                    log_dir = os.path.dirname(path)
-                    if not os.path.exists(log_dir):
-                        os.mkdir(log_dir)
-                    with open(os.path.join(log_dir, os.path.basename(path)), "wb") as f:
-                        f.write(file_bytes)
-            else:
-                raise Exception("There are no Tensorboard files in the array!")
+            tb_contents = model_array[0:tensorboard_size]["tensorboard"]
+            tensorboard_files = pickle.loads(tb_contents.tobytes())
+
+            for path, file_bytes in tensorboard_files.items():
+                log_dir = os.path.dirname(path)
+                if not os.path.exists(log_dir):
+                    os.mkdir(log_dir)
+                with open(os.path.join(log_dir, os.path.basename(path)), "wb") as f:
+                    f.write(file_bytes)
+
+    @staticmethod
+    def is_v1(array: tiledb.Array) -> bool:
+        if array.schema.domain.size < np.iinfo(np.uint64).max - 1025:
+            return True
+        return False
```

Comment:
Can be oneline: `return array.schema.domain.size < np.iinfo(np.uint64).max - 1025`